src/lib/storage/buffer/frame.hpp

```diff
@@ -35,73 +46,129 @@ namespace hyrise {
  *                                                                                    ^                                     ^ unlock_shared   |    |
  *                                                                                    | unlock_shared                       +-----------------+    |
  *                                                                                    |                                                            |
- *                                                                                    |                                                            |
+ *                                                                                    | (only if no other latches are present)                     |
  *                                                                                    +------------------------------------------------------------+
- * Graphic was created with https://github.com/ggerganov/dot-to-ascii
-*/
-
-class Frame final {
+ */
+class Frame final : private Noncopyable {
  public:
+  // Type of state and version
   using StateVersionType = uint64_t;
 
-  // State constants
+  // UNLOCKED has a value of 0
   static constexpr StateVersionType UNLOCKED = 0;
-  static constexpr StateVersionType LOCKED_SHARED = 0xFFFF - 3;  // 252 if 8 bits, 65532
-  static constexpr StateVersionType LOCKED = 0xFFFF - 2;         // 253 if 8 bits, 65533
-  static constexpr StateVersionType MARKED = 0xFFFF - 1;         // 254 if 8 bits, 65534
-  static constexpr StateVersionType EVICTED = 0xFFFF;            // 255 if 8 bits, 65535 for 16 bits
 
+  // SINGLE_LOCKED_SHARED signifies that the frame has a single reader. It has a value of 1. Used for convenience.
+  static constexpr StateVersionType SINGLE_LOCKED_SHARED = 1;
+
+  // LOCKED_SHARED signifies that the frame has the maxium amount of shared readers. It has a value of 65532.
```

Comment:
`LOCK_SHARED` in the comment vs `MAX_LOCKED_SHARED` as the constant name might be misleading. In all the comments, I think the constant name does not need to be repeated. E.g., `The maximum number of shared readers. Value: 65532` is sufficient, in my opinion.